### PR TITLE
posthog/sqlcommenter: middleware

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: posthog/sqlcommenter.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+class QueryWrapper:
+    def __init__(self, request):
+        self.request = request
+
+    def __call__(self, execute, sql, params, many, context):
+
+        resolver_match = self.request.resolver_match
+
+        sql_comment = generate_sql_comment(
```
Comment (macobo) [inline posthog/sqlcommenter.py:59]: Let's use the same pattern here as we do for clickhouse comments for consistency.

**Comment by macobo** [comment]
Related PR: https://github.com/PostHog/posthog/pull/7486/files

This got reverted due to some recursion errors due to sentry also hooking in in weird ways.

File: posthog/sqlcommenter.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            # See https://docs.djangoproject.com/en/stable/ref/urlresolvers/#django.urls.ResolverMatch.route
+
+            route=(resolver_match.route or None) if resolver_match else None,
+
+            # app_name is the application namespace for the URL pattern that matches the URL.
+            # See https://docs.djangoproject.com/en/stable/ref/urlresolvers/#django.urls.ResolverMatch.app_name
+            app_name=(resolver_match.app_name or None) if resolver_match else None,
+        )
+
+        sql += sql_comment
```
[the remark this answers is not in the document]
Comment (guidoiaquinti) [reply posthog/sqlcommenter.py:72]: I've used both in the past and I personally prefer to have the metadata as suffix

[the remark this answers is not in the document]
Comment (macobo) [reply posthog/sqlcommenter.py:72]: Should we update clickhouse commenting then as well for consistency?

**Comment by weyert** [comment]
As I am part of the project I would like to ask, what stopped you from using: https://github.com/open-telemetry/opentelemetry-sqlcommenter/tree/main/python/sqlcommenter-python

**Comment by guidoiaquinti** [comment]
> As I am part of the project I would like to ask, what stopped you from using: https://github.com/open-telemetry/opentelemetry-sqlcommenter/tree/main/python/sqlcommenter-python

👋 Hi @weyert ! This is a draft PR so I was simply prototyping but as you can see from the code I took inspiration from https://github.com/google/sqlcommenter/tree/master/python/sqlcommenter-python (that looks like a wrapper on top of the opentelemetry commenter you've linked).

**Comment by weyert** [comment]
> > As I am part of the project I would like to ask, what stopped you from using: https://github.com/open-telemetry/opentelemetry-sqlcommenter/tree/main/python/sqlcommenter-python
> 
> 👋 Hi @weyert ! This is a draft PR so I was simply prototyping but as you can see from the code I took inspiration from https://github.com/google/sqlcommenter/tree/master/python/sqlcommenter-python (that looks like a wrapper on top of the opentelemetry commenter you've linked).

Google's SQLCommenter got merged into the Opentelemetry project :)
See also: https://cloud.google.com/blog/products/databases/sqlcommenter-merges-with-opentelemetry

**Pull request closed without merging**